common/templates/context_funcs.go

```diff
@@ -1491,6 +1494,164 @@ func (c *Context) tmplThreadMemberRemove(threadID, memberID interface{}) string
 	return ""
 }
 
+func ConvertTagNameToId(c *dstate.ChannelState, tagName string) int64 {
+
+	if c.AvailableTags == nil {
+		return 0
+	}
+
+	// walk available tags list and see if there's a match
+	for _, tag := range c.AvailableTags {
+		if tag.Name == tagName {
+			return tag.ID
+		}
+	}
+
+	return 0
+}
+
+func ProcessOptionalForumPostArgs(c *dstate.ChannelState, values ...interface{}) (int, []int64, error) {
+
+	if values == nil || len(values) == 0 {
+		return c.DefaultThreadRateLimitPerUser, nil, nil
+	}
+
+	threadSdict, err := StringKeyDictionary(values...)
+	if err != nil {
+		return 0, nil, err
+	}
+
+	rateLimit := c.DefaultThreadRateLimitPerUser
+	var tags []int64 = nil
+	for key, val := range threadSdict {
+
+		key = strings.ToLower(key)
+		switch key {
+		case "slowmode":
+			rateLimit = tmplToInt(val)
+		case "tags":
+			if c.AvailableTags == nil {
+				break
+			}
+
+			v, _ := indirect(reflect.ValueOf(val))
+			const maxTags = 5 // discord limit
+			if v.Kind() == reflect.String {
+				tag := ConvertTagNameToId(c, ToString(val))
+				// ensure supplied id is valid
+				if tag > 0 {
+					tags = []int64{tag}
+				}
+			} else if v.Kind() == reflect.Slice {
+				// used to get rid of any duplicate tags the user might have sent
+				seen := make(map[string]struct{})
+				size := v.Len()
+				if size > maxTags {
+					size = maxTags
+				}
+
+				tags = make([]int64, 0, size)
+				for i := 0; i < v.Len() && len(seen) < size; i++ {
+					name := ToString(v.Index(i).Interface())
+					if len(name) == 0 {
+						continue
+					}
+
+					_, ok := seen[name]
+					if ok {
+						continue
+					}
+
+					// try to convert and check if the id is valid
+					tag := ConvertTagNameToId(c, name)
+					if tag == 0 {
+						continue
+					}
+
+					seen[name] = struct{}{}
+					tags = append(tags, tag)
+				}
+
+			} else {
+				return 0, nil, errors.New("tags must be of type string or cslice")
+			}
+		default:
+			return 0, nil, errors.New(`invalid key "` + key + `"`)
+		}
+	}
+
+	return rateLimit, tags, nil
+}
+
+func (c *Context) tmplCreateForumPost(channel, content interface{}, name string, optional ...interface{}) (*CtxChannel, error) {
```

Comment:
To me workflow > channel, name, content, (optional...) feels better. Post title is sort of first than post content. I know it needs interface{} twice then : )

Reply:
Latest commit switches the order to be channel, name, content, (optional...)

Reply:
I'm also wondering if `createThread` from the last PR should be changed to follow the same format. For example 
createThread channel name (optional: "message_id" arg "private" arg)

Reply:
looks reasonable, to have thread's name first rest is already all content and how content is shown

Reply:
it think that parameter `name` could also be part of type interface{} and you later ToString() it, because in some cases it is easier to use pure numbers as thread name : )

Reply:
Pushed a change that switches both create functions to use interface{} for name. It calls ToString() on it before passing it to discordgo.